src/player.cpp

```diff
@@ -11168,12 +11179,25 @@ activate your weapon."), gun->tname().c_str(), _(mod->location.c_str()));
 
         if (choice < int(mods.size())) {
             const std::string mod = used->contents[choice].tname();
-            remove_gunmod(used, unsigned(choice));
-            add_msg(_("You remove your %s from your %s."), mod.c_str(), used->tname().c_str());
+
+            if( used->contents[choice].has_flag("IRREMOVABLE") ){
+                add_msg(_("You cannot remove integrated mods.") );
+            } else{
+                remove_gunmod(used, unsigned(choice));
+                add_msg(_("You remove your %s from your %s."), mod.c_str(), used->tname().c_str());
+            }
         } else if (choice == int(mods.size())) {
+            bool irremovable = false;
             for (int i = used->contents.size() - 1; i >= 0; i--) {
-                remove_gunmod(used, i);
+                if( used->contents[i].has_flag("IRREMOVABLE") ){
+                    irremovable = true;
+                }
+                else{
```

Comment:
I suppose I could do that, was testing around with some files locally but didn't want to push it as i was modifying perfectly fine guns. Ill also get to fixing those code quality issues.